### ISSUE TITLE
Training the model further - degrades model very quickly!

### ISSUE THREAD

**Issue by jackureu** [CSAILVision/semantic-segmentation-pytorch#97]
I took the pretrained model (resnet101-upernet) and tried to train it further. Without changing anything. Same dataset, same parameters:

user@machine:~/projects/semantic_segmentation_pytorch$ python3 ./train.py --num_gpus 4 --arch_encoder resnet101 --arch_decoder upernet --fc_dim 2048 --id user_model --segm_downsampling_rate 4 --padding_constant 32 --weights_encoder ./baseline-resnet101-upernet/encoder_epoch_25.pth --weights_decoder ./baseline-resnet101-upernet/decoder_epoch_25.pth --start_epoch 25 --num_epoch 30

But the model seems to quickly degrade in performance:

Input arguments:
padding_constant 32
fc_dim           2048
deep_sup_scale   0.4
num_class        150
weights_encoder  ./baseline-resnet101-upernet/encoder_epoch_25.pth
num_gpus         4
random_flip      True
start_epoch      25
segm_downsampling_rate 4
list_val         ./data/validation.odgt
arch_encoder     resnet101
ckpt             ./ckpt
epoch_iters      5000
arch_decoder     upernet
workers          16
lr_encoder       0.02
seed             304
optim            SGD
beta1            0.9
imgMaxSize       1000
lr_decoder       0.02
lr_pow           0.9
weights_decoder  ./baseline-resnet101-upernet/decoder_epoch_25.pth
weight_decay     0.0001
list_train       ./data/train.odgt
disp_iter        20
num_epoch        30
id               user_model
imgSize          [300, 375, 450, 525, 600]
batch_size_per_gpu 2
root_dataset     ./data/
fix_bn           0
Model ID: user_model-resnet101-upernet-ngpus4-batchSize8-imgMaxSize1000-paddingConst32-segmDownsampleRate4-LR_encoder0.02-LR_decoder0.02-epoch30-decay0.0001-fixBN0
Loading weights for net_encoder
Finished loading weights for net_encoder
Loading weights for net_decoder
Finished loading weights for net_decoder
# samples: 20210
1 Epoch = 5000 iters
Num gpus: 4
Epoch: [25][0/5000], Time: 13.61, Data: 0.32, lr_encoder: 0.020000, lr_decoder: 0.020000, Accuracy: 95.36, Loss: 0.134241
Epoch: [25][20/5000], Time: 5.66, Data: 0.02, lr_encoder: 0.004696, lr_decoder: 0.004696, Accuracy: 95.26, Loss: 0.134772
Epoch: [25][40/5000], Time: 5.43, Data: 0.01, lr_encoder: 0.004693, lr_decoder: 0.004693, Accuracy: 94.59, Loss: 0.157938
Epoch: [25][60/5000], Time: 5.36, Data: 0.01, lr_encoder: 0.004690, lr_decoder: 0.004690, Accuracy: 93.55, Loss: 0.191192
Epoch: [25][80/5000], Time: 5.28, Data: 0.01, lr_encoder: 0.004687, lr_decoder: 0.004687, Accuracy: 92.49, Loss: 0.223331
Epoch: [25][100/5000], Time: 5.19, Data: 0.01, lr_encoder: 0.004685, lr_decoder: 0.004685, Accuracy: 90.83, Loss: 0.291658
Epoch: [25][120/5000], Time: 5.15, Data: 0.01, lr_encoder: 0.004682, lr_decoder: 0.004682, Accuracy: 87.99, Loss: 0.415997
Epoch: [25][140/5000], Time: 5.15, Data: 0.00, lr_encoder: 0.004679, lr_decoder: 0.004679, Accuracy: 84.12, Loss: 0.603035
Epoch: [25][160/5000], Time: 5.15, Data: 0.00, lr_encoder: 0.004676, lr_decoder: 0.004676, Accuracy: 79.89, Loss: 0.796039
Epoch: [25][180/5000], Time: 5.12, Data: 0.00, lr_encoder: 0.004673, lr_decoder: 0.004673, Accuracy: 76.87, Loss: 0.933330
Epoch: [25][200/5000], Time: 5.11, Data: 0.00, lr_encoder: 0.004670, lr_decoder: 0.004670, Accuracy: 74.34, Loss: 1.039475
Epoch: [25][220/5000], Time: 5.10, Data: 0.00, lr_encoder: 0.004668, lr_decoder: 0.004668, Accuracy: 72.51, Loss: 1.106258
Epoch: [25][240/5000], Time: 5.11, Data: 0.00, lr_encoder: 0.004665, lr_decoder: 0.004665, Accuracy: 71.57, Loss: 1.141811
Epoch: [25][260/5000], Time: 5.10, Data: 0.00, lr_encoder: 0.004662, lr_decoder: 0.004662, Accuracy: 70.86, Loss: 1.162392
Epoch: [25][280/5000], Time: 5.08, Data: 0.00, lr_encoder: 0.004659, lr_decoder: 0.004659, Accuracy: 70.27, Loss: 1.177229
Epoch: [25][300/5000], Time: 5.08, Data: 0.00, lr_encoder: 0.004656, lr_decoder: 0.004656, Accuracy: 69.86, Loss: 1.188734

The only difference I can think of is that my machine has 4 GPUs instead of the 8 that were used during the original training... But the difference seems really big...

Any ideas?

**Comment by jackureu** [comment]
For the sake of completeness- I'm using Python 3.5.2 with torch 0.4.1 and numpy 1.15.0

And I'm getting the following errors when executing train.py (It seems that non of them is important):

/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/home/user/projects/semantic_segmentation_pytorch/models/models.py:67: UserWarning: nn.init.kaiming_normal is now deprecated in favor of nn.init.kaiming_normal_.
  nn.init.kaiming_normal(m.weight.data)
/home/user/.local/lib/python3.5/site-packages/torch/nn/functional.py:1961: UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
/home/user/.local/lib/python3.5/site-packages/torch/nn/functional.py:1890: UserWarning: nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.")
./train.py:51: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  ave_total_loss.update(loss.data[0])
./train.py:52: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  ave_acc.update(acc.data[0]*100)
./train.py:66: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  history['train']['loss'].append(loss.data[0])
./train.py:67: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  history['train']['acc'].append(acc.data[0])

**Comment by hangzhaomit** [comment]
First, you should use a much smaller learning rate to finetune the model;
Second, as you do not have 8 GPUs, it is better to freeze the BNs.
